### PR TITLE
made code examples user friendly; fixes #7720 [WIP]

### LINKED ISSUE #7720: kivy.org code examples are not user friendly
https://kivy.org/doc/stable/tutorials/firstwidget.html
you cant just copy the code snippets without  line numbers.

### PR DESCRIPTION
**Pull request by p0lygun**
Currently when copying from a `include` source code the line numbers are copied, this PR fixes that 
## current 
image
## After fix
image

Maintainer merge checklist
* [x] Title is descriptive/clear for inclusion in release notes.
* [ ] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [ ] Added to the milestone version it was merged into.
* [ ] **Unittests** are included in PR.
* [ ] Properly documented, including `versionadded`, `versionchanged` as needed.

### CODE REVIEW

**Review by misl6 — COMMENTED** [review]
`user-select` only disables selection, but unfortunately doesn't prevent copying. (At least on Webkit browsers)

File: doc/sources/.static/kivy.js [hunk 1]
```diff
@@ -224,5 +224,12 @@ $(function() {
...
 	}
-
+	// hack to add a attrib to all linenos
+    $(".linenos").each(function(){
+        let $this = $(this);
+        $this.attr("data-line-number", $this.text());
+		$this.contents().filter(function(){
+    			return this.nodeType === 3; //https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType
+		}).remove();
+  });
```
Comment (misl6) [inline doc/sources/.static/kivy.js:234]: Not a super-fan of this kind of hack. Maybe we will change it in the future,  but right now, does its job (at least on `kivy.org`), so I'm fine with it.

File: doc/sources/.static/kivy.js [hunk 2]
```diff
@@ -224,5 +224,12 @@ $(function() {
 		var section_title = $('li.toctree-l1.current > a').text();
 		$('div.body h1:eq(0)').prepend(section_title + ' &raquo; ');
 	}
-
+	// hack to add a attrib to all linenos
```
Comment (misl6) [inline doc/sources/.static/kivy.js:227]: Can we clarify why is that needed?

File: doc/sources/.static/fresh.css [hunk 3]
```diff
@@ -635,6 +635,15 @@ code.docutils{
...
 }
+.linenos::before {
+	line-height: 0;
+	font-size: 14px;
+	content: attr(data-line-number);
+}
+.linenos {
+	font-size: 0;
+	line-height: 80%;
+}
```
Comment (misl6) [inline doc/sources/.static/fresh.css:646]: Is really needed?

File: doc/sources/.static/fresh.css [hunk 4]
```diff
@@ -635,6 +635,15 @@ code.docutils{
 	}
 
 }
+.linenos::before {
+	line-height: 0;
+	font-size: 14px;
```
Comment (misl6) [inline doc/sources/.static/fresh.css:640]: Are `line-height` and `font-size` needed?

File: doc/sources/.static/fresh.css [hunk 5]
```diff
@@ -635,6 +635,15 @@ code.docutils{
 	}
 
 }
+.linenos::before {
```
Comment (misl6) [inline doc/sources/.static/fresh.css:638]: Can be moved on top of the `@media` queries? (Just after `code.docutils`)

**Review by misl6 — CHANGES_REQUESTED** [review]
Tested on Safari, Chrome, Firefox and the hack works as expected.
I've added some comments.

File: doc/sources/.static/kivy.js (hunk 2 above)
Comment (p0lygun) [reply doc/sources/.static/kivy.js:227]: the comment ?

File: doc/sources/.static/fresh.css (hunk 3 above)
Comment (p0lygun) [reply doc/sources/.static/fresh.css:646]: wait actually no, as there is no text in linenos now i'll remove it

File: doc/sources/.static/kivy.js (hunk 2 above)
Comment (misl6) [reply doc/sources/.static/kivy.js:227]: Nope, the comment is fine, just needs to be a little bit verbosely about it.
As an example (but feel free to change it): "Hack to add an attrib to all linenos, this is needed in order to avoid that certain browsers (Ex: Safari) treat the line-number as a copyable item."

Comment (p0lygun) [reply doc/sources/.static/kivy.js:227]: ahh i see u mean it should be more descriptive, correct

**Review by misl6 — APPROVED** [review]
